src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/RestoreTelemetryEvent.cs

```diff
@@ -31,6 +31,7 @@ public class RestoreTelemetryEvent : ActionEventBase
         public const string NumLocalFeeds = nameof(NumLocalFeeds);
         public const string NuGetOrg = nameof(NuGetOrg);
         public const string VsOfflinePackages = nameof(VsOfflinePackages);
+        public const string ExplicitRestoreReason = nameof(ExplicitRestoreReason);
```

Comment:
Please add a test to prove that this Property is sent when we have explicit restores.

Reply:
You want me to test the telemetry event or a restore? Explicit restores are done in the UI, not sure if it is easy to test that, I'll try

Reply:
Just telemetry event. Testin a whole restore operation takes time.

You can modify existing tests to prove that property is set. For example:

https://github.com/NuGet/NuGet.Client/blob/a5eb6986236572b4322813b0231d3ed2e2038e5b/test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Telemetry/RestoreTelemetryServiceTests.cs#L29